packages/xo-server/src/xo-mixins/subjects.mjs

```diff
@@ -365,7 +365,18 @@ export default class {
 
     if (name) group.name = name
 
-    await this._groups.update(group)
+    try {
+      await this._groups.update(group)
+    } catch (error) {
+      if (error.message === `the group ${name} already exists`) {
+        const existingGroup = await this._groups.get({ name })
+        throw objectAlreadyExists({
+          objectId: existingGroup?.id,
```

Comment:
Why `?`? if you get an error because the group already exists, `existingGroup` is not undefined.